server/src/main/java/org/elasticsearch/action/support/replication/PostWriteRefresh.java

```diff
@@ -24,11 +25,14 @@
 
 public class PostWriteRefresh {
 
+    public static final String POST_WRITE_REFRESH_ORIGIN = "post_write_refresh";
     public static final String FORCED_REFRESH_AFTER_INDEX = "refresh_flag_index";
     private final TransportService transportService;
+    private final ThreadPool threadPool;
 
-    public PostWriteRefresh(final TransportService transportService) {
+    public PostWriteRefresh(final TransportService transportService, final ThreadPool threadPool) {
```

Comment:
Instead of this there is a public method to get the `ThreadPool` from the `TransportService`. If for some reason I am missing something and that does not work, I would just prefer to pass the `ThreadContext` down instead of the whole thread pool.

Reply:
Thanks @Tim-Brooks for the suggestion! I thought it was another thread pool,  but it seems to work so far. It simplifies the PR a lot.